test/controllers/drivers_controller_test.rb

```diff
@@ -5,68 +5,114 @@
 
   describe "index" do
     it "responds with success when there are many drivers saved" do
-      # Arrange
-      # Ensure that there is at least one Driver saved
-
+      Driver.create(name: "CheezitMan", vin: "ABCDEFGHIJKLMNOPQ")
       # Act
-
+      get drivers_path
       # Assert
-
+      must_respond_with :success
     end
 
     it "responds with success when there are no drivers saved" do
       # Arrange
       # Ensure that there are zero drivers saved
 
       # Act
-
+      get drivers_path
       # Assert
-
+      must_respond_with :success
     end
   end
 
   describe "show" do
     it "responds with success when showing an existing valid driver" do
+      driver = Driver.create(name: "CheezitMan", vin: "ABCDEFGHIJKLMNOPQ")
       # Arrange
-      # Ensure that there is a driver saved
 
+      # Ensure that there is a driver saved
       # Act
-
+      get driver_path(driver.id)
       # Assert
-
+      must_respond_with :success
     end
 
     it "responds with 404 with an invalid driver id" do
       # Arrange
       # Ensure that there is an id that points to no driver
 
       # Act
-
+      get driver_path(Driver.last.id + 1)
       # Assert
-
+      must_respond_with :not_found
     end
   end
 
   describe "new" do
     it "responds with success" do
+      #arrange
+      get new_driver_path
+      #Assert
+      must_respond_with :success
     end
   end
 
   describe "create" do
     it "can create a new driver with valid information accurately, and redirect" do
       # Arrange
       # Set up the form data
-
+      driver_hash = {
+          driver: {
+              name: "Solomon Mehru",
+              vin: "WBWSS52P9NEYLVDE9",
+          },
+      }
       # Act-Assert
-      # Ensure that there is a change of 1 in Driver.count
+      #Ensure that there is a change of 1 in Driver.count
+      expect {
+              post drivers_path, params: driver_hash
+            }.must_change "Driver.count", 1
+
+            new_driver = Driver.find_by(name: driver_hash[:driver][:name])
+            expect(new_driver.vin).must_equal driver_hash[:driver][:vin]
+            # expect(new_driver.available).must_equal :true
+
+            must_respond_with :redirect
+            must_redirect_to driver_path(new_driver.id)
 
       # Assert
       # Find the newly created Driver, and check that all its attributes match what was given in the form data
       # Check that the controller redirected the user
+      end
 
+
+    it "does not create a driver if a proper vin is not provided" do
+
+      driver_hash = {
+          driver: {
+              name: "Richard Salazar",
+              vin: "48765"
+          }
+      }
+
+      #Assert
+      expect {
```

Comment:
You should also include expectations for the response code.  Later with flash notices (not in this project) also test the error messages.